props/saber_BC_buttons.h

```diff
@@ -382,40 +1148,252 @@ EFFECT(volup);      // for increse volume
-#ifdef ENABLE_FASTON
-#warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
-EFFECT(faston);     // for EFFECT_FAST_ON. Being replaced by fastout.wav, which is already defined in the main OS.
-#endif
-EFFECT(blstbgn);    // for Begin Multi-Blast
-EFFECT(blstend);    // for End Multi-Blast
 EFFECT(push);       // for Force Push gesture
-EFFECT(monosfx);    // for Monophonically played sounds (iceblade, seismic charge etc...)
-EFFECT(swap);       // for standalone triggering EffectSequence<>
+EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
+EFFECT(mzoom);      // for Spam Blast enter/exit
+
+template<class SPEC>
+struct BCVolumeMode : public SPEC::SteppedMode {
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int percentage_ = 0;
+
+  int steps_per_revolution() override {
+    return 12;  // adjust for sensitivity
+  }
+
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    mode::getSL<SPEC>()->SayEnterVolumeMenu();
+    announce_volume();
+    SPEC::SteppedMode::mode_activate(onreturn);
+  }
+
+  void announce_volume() {
+    if (percentage_ <= 10) {
+      mode::getSL<SPEC>()->SayMinimumVolume();
+    } else if (percentage_ >=100) {
+      mode::getSL<SPEC>()->SayMaximumVolume();
+    } else {
+      mode::getSL<SPEC>()->SayWhole(percentage_);
+      mode::getSL<SPEC>()->SayPercent();
+    }
+  }
+
+  void mode_deactivate() override {
+    announce_volume();
+    mode::getSL<SPEC>()->SayVolumeMenuEnd();
+    SPEC::SteppedMode::mode_deactivate();
+  }
+
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
+      if (current_volume_ >= max_volume_) {
+        current_volume_ = max_volume_;
+        QuickMaxVolume();
+      } else {
+        mode::getSL<SPEC>()->SayVolumeUp();
+      }
+      dynamic_mixer.set_volume(current_volume_);
+    }
+  }
+
+  void prev() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ > min_volume_) {
+      current_volume_ -= max_volume_ * 0.10;
+      if (current_volume_ <= min_volume_) {
+        current_volume_ = min_volume_;
+        QuickMinVolume();
+      } else {
+        mode::getSL<SPEC>()->SayVolumeDown();
+      }
+      dynamic_mixer.set_volume(current_volume_);
+    }
+  }
+
+  void QuickMaxVolume() {
+    dynamic_mixer.set_volume(max_volume_);
+    PVLOG_NORMAL << "** Maximum Volume\n";
```

Comment:
Seems like a lot of these could be PVLOG_VERBOSE

Reply:
Could these stay just until finished testing?
Then one clean, small PR could be raising the print log level.
Other verbose messages really clog up the console.

Reply:
Maybe PVLOG_DEBUG ?

Reply:
As mentioned above, for now while still testing this PR, it's helpful to have these show without all the other system messages that print with PROFFIEOS_LOG_LEVEL higher than 300.
My intent is to flip most if not all of these printouts to a higher level PVLOG as one of the last pushes.